pkg/kf/commands/config/config.go

```diff
@@ -30,16 +34,83 @@ import (
 	scv1beta1 "github.com/poy/service-catalog/pkg/client/clientset_generated/clientset/typed/servicecatalog/v1beta1"
 	"github.com/poy/service-catalog/pkg/svcat"
 	servicecatalog "github.com/poy/service-catalog/pkg/svcat/service-catalog"
+	"gopkg.in/yaml.v2"
 	k8sclient "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
 // KfParams stores everything needed to interact with the user and Knative.
 type KfParams struct {
-	Namespace   string
-	KubeCfgFile string
-	Verbose     bool
+	// Config holds the path to the configuration.
+	Config string `yaml:"-"`
+
+	// Namespace holds the namespace kf should connect to by default.
+	Namespace string `yaml:"space"`
+
+	// KubeCfgFile holds the path to the kubeconfig.
+	KubeCfgFile string `yaml:"kubeconfig"`
+}
+
+// ConfigPath gets the path we should read the config from/write it to.
+func (p *KfParams) ConfigPath() string {
+	if p.Config != "" {
+		return p.Config
+	}
+
+	// Kf shouldn't fail if we can't find the user's home directory, instead
+	// use the current working directory.
+	base := "."
+	if home, err := homedir.Dir(); err == nil {
+		base = home
+	}
+
+	return path.Join(base, ".kf")
+}
+
+// ReadConfig reads the config from the specified config path or the default
+// path. If the path is the default and the file doesn't yet exist, then
+// this function does nothing.
+func (p *KfParams) ReadConfig() error {
+	userSpecifiedConfig := p.Config != ""
+	configPath := p.ConfigPath()
+
+	contents, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		switch {
+		case userSpecifiedConfig:
+			return err
+		case os.IsNotExist(err):
+			return nil
+		default:
+			return err
+		}
+	}
+
+	newParams := KfParams{}
+	if err := yaml.Unmarshal(contents, &newParams); err != nil {
+		return err
+	}
+
+	return p.ApplyDefaults(newParams)
+}
+
+// ApplyDefaults changes blank fields in this struct to the value in defaults.
+func (p *KfParams) ApplyDefaults(defaults KfParams) error {
```

Comment:
I'm not sure this method should have input parameters, but should instead rely on const values defined in this file.

Furthermore, I'm not sure this method should be exported.